sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleService.java

```diff
@@ -725,44 +725,37 @@ public Predicate buildCriteriaFilter(SampleCriteria criteria, SampleQueryContext
 		} else if (sampleAssociationType == SampleAssociationType.EVENT_PARTICIPANT) {
 			filter = CriteriaBuilderHelper.and(cb, filter, cb.isNotNull(joins.getEventParticipant()));
 		}
-
 		if (criteria.getRegion() != null) {
-			Expression<Object> regionExpression = cb.selectCase()
-				.when(cb.isNotNull(joins.getCaseRegion()), joins.getCaseRegion().get(AbstractDomainObject.UUID))
-				.otherwise(
-					cb.selectCase()
-						.when(cb.isNotNull(joins.getContactRegion()), joins.getContactRegion().get(AbstractDomainObject.UUID))
-						.otherwise(
-							cb.selectCase()
-								.when(cb.isNotNull(joins.getContactCaseRegion()), joins.getContactCaseRegion().get(AbstractDomainObject.UUID))
-								.otherwise(joins.getEventRegion().get(AbstractDomainObject.UUID))));
+			final String regionUuid = criteria.getRegion().getUuid();
 			filter = CriteriaBuilderHelper.and(
 				cb,
 				filter,
-				cb.or(
-					cb.and(
-						cb.isNotNull(joins.getCaze()),
-						cb.equal(joins.getCaseResponsibleRegion().get(AbstractDomainObject.UUID), criteria.getRegion().getUuid())),
-					cb.equal(regionExpression, criteria.getRegion().getUuid())));
+				CriteriaBuilderHelper.or(
+					cb,
+					cb.equal(joins.getCaseRegion().get(AbstractDomainObject.UUID), regionUuid),
```

Comment:
Could you use `Region.UUID?` instead of `AbstractDomainObject.UUID`, it's makes it more obvious which entity is currently in focus. Same for District and Community where the pattern is the same. See also the other services.